esp/esp/program/models/class_.py

```diff
@@ -1111,8 +1125,36 @@ def firstBlockEvent(self):
     def isFull(self, ignore_changes=False, webapp=False):
         if len(self.get_meeting_times()) == 0:
             return True
-        if webapp and Tag.getBooleanTag('count_checked_in_only', program = self.parent_program, default = False):
+
+        # Get time and tag values to determine what number to base class changes on
+        now = datetime.datetime.now()
+        switch_time = None
+        if Tag.getProgramTag('switch_time_program_attendance', program=self.parent_program):
+            try:
+                switch_time_str = now.strftime("%Y/%m/%d ") + Tag.getProgramTag('switch_time_program_attendance', program=self.parent_program)
+                switch_time = datetime.datetime.strptime(switch_time_str, "%Y/%m/%d %H:%M")
+            except ValueError:
+                pass
+        switch_lag = None
+        if Tag.getProgramTag('switch_lag_class_attendance', program = self.parent_program):
+            try:
+                switch_lag = int(Tag.getProgramTag('switch_lag_class_attendance', program = self.parent_program))
+            except ValueError:
+                pass
+
+        # Mode 1: Base "fullness" on class attendance numbers if:
+        # 1) using webapp, 2) 'switch_lag_class_attendance' tag is set properly
+        # 3) it is currently past the class start time + however many minutes specified in tag
+        # 4) at least one student has been marked as attending the class
+        if webapp and switch_lag and now >= (self.start_time_prefetchable() + timedelta(minutes=switch_lag)) and self.num_students_attending() >= 1:
+            num_students = self.num_students_attending()
+        # Mode 2: Base "fullness" on program attendance numbers if:
+        # 1) using webapp, 2) 'switch_time_program_attendance' tag is set properly
+        # 3) it is currently past the time specified in tag
+        # 4) at least one student has been marked as attending the program
```

Comment:
Happy to change condition 4 to be at least 5 students attending the program if that seems good in order to avoid problems with test accounts?